sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c

```diff
@@ -115,7 +115,7 @@ gnrc_pktsnip_t *gnrc_pktbuf_mark(gnrc_pktsnip_t *pkt, size_t size, gnrc_nettype_
     mutex_lock(&gnrc_pktbuf_mutex);
     if ((size == 0) || (pkt == NULL) || (size > pkt->size) || (pkt->data == NULL)) {
         DEBUG("pktbuf: size == 0 (was %" PRIuSIZE ") or pkt == NULL (was %p) or "
-              "size > pkt->size (was %" PRIuSIZE ") or pkt->data == NULL (was %p)\n",
```

Comment:
`%u` is for `unsigned int`, which may not be the same as `uint16_t`, e.g. on ARM. Yes, there is integer promotion, but that may also promote to `signed int`, which if that were guaranteed to be two's complement would be bitwise identical with the corresponding `unsigned int` for the value range, but the C standard chose to net mandate two's complement.

So yada yada please add a cast to `unsigned` in front of `pkt->size`.

Reply:
That's great news! 🎉